app/parse.py

```diff
@@ -15,8 +39,106 @@ class Product:
     num_of_reviews: int
 
 
+def get_product(product_card: WebElement) -> Product:
+    return Product(
+        title=product_card.find_element(By.CLASS_NAME, "title").get_attribute("title"),
+        description=product_card.find_element(By.CLASS_NAME, "description").text,
+        price=float(
+            product_card.find_element(By.CLASS_NAME, "price").text.replace("$", "")
+        ),
+        rating=len(product_card.find_elements(By.CLASS_NAME, "ws-icon-star")),
+        num_of_reviews=int(
+            product_card.find_element(By.CLASS_NAME, "review-count").text.split()[0]
+        ),
+    )
+
+
+def get_all_from_dynamic_page(url: str) -> list[Product]:
+    logging.info(f"Get all products dynamic page'{url}'")
+    driver = get_driver()
+    driver.get(urljoin(BASE_URL, url))
+    try:
+        close_button = driver.find_element(By.CLASS_NAME, "acceptCookies")
+        driver.execute_script("arguments[0].scrollIntoView(true);", close_button)
+        close_button.click()
+        logging.info("Closed cookie banner")
+    except NoSuchElementException:
+        logging.info("Cookie banner not found")
+    except ElementNotInteractableException:
+        logging.info("The cookie banner was unclickable")
+    try:
+        more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+        while True:
+            try:
+                if not more.is_displayed():
+                    break
+                ActionChains(driver).move_to_element(more).click(more).perform()
+                logging.info("Clicked 'Load More' button")
+                time.sleep(1)
+                more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+            except ElementClickInterceptedException:
+                logging.warning(
+                    "Load More button was unclickable;"
+                    " retrying after closing any obstructive elements"
+                )
+                cookie_button = driver.find_element(By.CLASS_NAME, "acceptCookies")
+                if cookie_button:
+                    cookie_button.click()
+                time.sleep(1)
+    except NoSuchElementException:
+        logging.info("This page is static")
+    cards = driver.find_elements(By.CLASS_NAME, "product-wrapper")
+    return [get_product(card) for card in cards]
+
+
 def get_all_products() -> None:
-    pass
+    driver = WebDriver()
```

Comment:
The instantiation of `WebDriver` here is incorrect. You need to specify the driver type, such as `webdriver.Chrome()` or `webdriver.Firefox()`, and ensure the appropriate driver executable is available in your PATH or specified explicitly.